### ISSUE TITLE
When using CLI and JSON, sourceRoots requires the full path to the file

### ISSUE THREAD

**Issue by Zephyrus29** [Kotlin/dokka#2571]
- Operating system: Linux
- Build tool: Cli
- Dokka version: 1.7.10

When using the Dokka CLI interface, I've configured a json file but it requires the full path to be in the sourceRoots variable, a relative path does not work. I can copy the example, and it works, but changing to a relative path fails. 

Is this the intended behaviour?

**Comment by IgnatBeresnev** [comment]
Having looked at the code, I'd say it's a bug. 

I haven't tested it, but the problem is most likely in [DefaultDescriptorToDocumentableTranslator.kt#filterDescriptorsInSourceSet](https://github.com/Kotlin/dokka/blob/d61df34a784ab2f05d0facf0acad557b7743cf10/plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt#L144):
```
Paths.get(path).startsWith(root.toPath())
```

The code above probably needs to check that file with path
```
/home/ignat/IdeaProjects/dokka-debug-mvn/src/main/kotlin/org/jetbrains/dokka/Main.kt
```

startsWith

```
/home/ignat/IdeaProjects/dokka-debug-mvn/src/main/kotlin
```

I assume that's how it works for Gradle and Maven plugins, which pass absolute paths.

___

However, because JSON config is deserialized by Jackson as is, your relative path string `src/main/kotlin` ends up being `new File("src/main/kotlin")`.

If we try to apply the same logic in JShell, we'll see that the relative path is not resolved to absolute upon mapping it to `Path`:
```
jshell> new File("src/main/kotlin").toPath()
$4 ==> src/main/kotlin
```

So the check ends up being

```
/home/ignat/IdeaProjects/dokka-debug-mvn/src/main/kotlin/org/jetbrains/dokka/Main.kt
```

startsWith

```
src/main/kotlin
```

which is naturally `false`.

___

However, if you get absolute file first, it should resolve the issue.

```
jshell> new File("src/main/kotlin").getAbsoluteFile().toPath()
$5 ==> /home/ignat/IdeaProjects/dokka-debug-mvn/src/main/kotlin
```

In other words, the code needs to be updated to account for relative paths. This case also needs to be covered by the integration test, and maybe a unit test if possible.
